internal/generator/vector/output/elasticsearch/elasticsearch.go

```diff
@@ -48,203 +43,55 @@ api_version = "v{{ .Version }}"
-    .message = encode_json(.structured)
-    del(.structured)
-  }
-`)
-	}
-
-	return Remap{
-		Desc:        "Set Elasticsearch index",
-		ComponentID: id,
-		Inputs:      helpers.MakeInputs(inputs...),
-		VRL:         strings.Join(helpers.TrimSpaces(vrls), "\n"),
-	}
-}
-
-func FlattenLabels(id string, inputs []string) Element {
-	return Remap{
-		ComponentID: id,
-		Inputs:      helpers.MakeInputs(inputs...),
-		VRL: strings.Join([]string{
-			viaq.VRLOpenShiftSequence,
-			viaq.VRLDedotLabels,
-			VRLFlattenLabels,
-			VRLPruneLabels,
-		}, "\n"),
-	}
-}
-
-const (
-	VRLFlattenLabels = `if exists(.kubernetes.labels) {
-  .kubernetes.flat_labels = []
-  for_each(object!(.kubernetes.labels)) -> |k,v| {
-    .kubernetes.flat_labels = push(.kubernetes.flat_labels, join!([string(k),"=",string!(v)]))
-  }
-}`
-	VRLPruneLabels = `if exists(.kubernetes.labels) {
-  exclusions = ["app_kubernetes_io_name", "app_kubernetes_io_instance", "app_kubernetes_io_version", "app_kubernetes_io_component", "app_kubernetes_io_part-of", "app_kubernetes_io_managed-by", "app_kubernetes_io_created-by"]
-  keep = {}
-  for_each(object!(.kubernetes.labels))->|k,v|{
-    if !includes(exclusions, k) {
-      .kubernetes.labels, err = remove(object!(.kubernetes.labels),[k],true)
-      if err != null {
-        log(err, level: "error")
-      }
-    }
-  }
-}`
-)
-
 func (e *Elasticsearch) SetCompression(algo string) {
 	e.Compression.Value = algo
 }
 
-func New(id string, o logging.OutputSpec, inputs []string, secret *corev1.Secret, strategy common.ConfigStrategy, op Options) []Element {
-	outputs := []Element{}
-
-	esIndexID := helpers.MakeID(id, "add_es_index")
-	dedotID := helpers.MakeID(id, "dedot_and_flatten")
-
+func New(id string, o obs.OutputSpec, inputs []string, secrets helpers.Secrets, strategy common.ConfigStrategy, op Options) []Element {
 	if genhelper.IsDebugOutput(op) {
 		return []Element{
-			SetESIndex(esIndexID, inputs, o, op),
-			FlattenLabels(dedotID, []string{esIndexID}),
-			Debug(id, helpers.MakeInputs([]string{dedotID}...)),
+			Debug(id, helpers.MakeInputs(inputs...)),
 		}
 	}
-	sink := Output(id, o, []string{dedotID}, secret, op)
+	sink := Output(id, o, inputs, secrets, op)
 	if strategy != nil {
 		strategy.VisitSink(sink)
 	}
-	request := common.NewRequest(id, strategy)
-	request.TimeoutSecs.Value = 2147483648
-	outputs = MergeElements(outputs,
-		[]Element{
-			SetESIndex(esIndexID, inputs, o, op),
-			FlattenLabels(dedotID, []string{esIndexID}),
-			sink,
-			common.NewAcknowledgments(id, strategy),
-			common.NewBatch(id, strategy),
-			common.NewBuffer(id, strategy),
-			request,
-		},
-
-		common.TLS(id, o, secret, op),
-		BasicAuth(id, o, secret),
-	)
+	outputs := []Element{
+		sink,
+		common.NewAcknowledgments(id, strategy),
+		common.NewBatch(id, strategy),
+		common.NewBuffer(id, strategy),
+		common.NewRequest(id, strategy),
+		tls.New(id, o.TLS, secrets, op),
+		BasicAuth(id, o, secrets),
```

Comment:
Should this be using `HTTPAuth` method from `common/auth/auth.go`?

Reply:
pushed into separate PR